src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Dialog.kt

```diff
@@ -20,45 +20,144 @@ package io.github.yuutoproject.yuutobot.commands
+        val now: Long = System.currentTimeMillis()
+
+        if (args.count() < 2) {
+            event.channel.sendMessage("This command requires two arguments : `dialog [background] <character> <text>` ([] is optional)").queue()
+            return
+        }
+
+        var character: String = args.removeFirst().toLowerCase()
+        val background: String
+
+        if (characters.contains(character)) {
+            background = "camp"
+        } else {
+            background = character
+            character = args.removeFirst().toLowerCase()
+        }
+
+        if (!backgrounds.contains(background)) {
+            event.channel.sendMessage("Sorry but I couldn't find `$background` as a location\nAvailable backgrounds are: $backgroundsString").queue()
+            return
+        }
+
+        if (!characters.contains(character)) {
+            event.channel.sendMessage("Sorry, but I don't think that `$character` is a character in Camp Buddy\nAvailable characters are: $charactersString").queue()
+            return
+        }
+
+        if (args.count() < 1) {
+            event.channel.sendMessage("Please provide a message to be written~!").queue()
+            return
+        }
+
+        val text: String = args.joinToString(" ").replace("/[‘’]/g".toRegex(), "'")
+
+        if (text.length > 120) {
+            event.channel.sendMessage("Sorry, the message limit is 120 characters <:hiroJey:692008426842226708>").queue()
+            return
+        }
+
+        if (
+            EMOTE_MENTIONS_REGEX.containsMatchIn(text) ||
+            EMOJI_REGEX.containsMatchIn(text) ||
+            NONASCII_REGEX.containsMatchIn(text)
+        ) {
+            event.channel.sendMessage("Sorry, I can't display emotes, mentions, or non-latin characters").queue()
+            return
+        }
 
         event.channel.sendTyping().queue()
 
         val json = "application/json; charset=utf-8".toMediaType()
         val body = """{
-                "background": "messhall",
-                "character": "knox",
-                "text": "${args.joinToString(" ").replace("\"", "\\\"")}"
+                "background": "$background",
+                "character": "$character",
+                "text": "$text"
             }""".trimIndent().toRequestBody(json)
 
         val request = Request.Builder()
             .url("https://kyuu.to/dialog")
             .post(body)
             .build()
 
-        moveThisToAStorageLocation.newCall(request).enqueue(object : OkHttp3Callback {
+        httpClient.newCall(request).enqueue(object : OkHttp3Callback {
             override fun onFailure(call: Call, e: IOException) {
-                TODO("Not yet implemented")
+                event.channel.sendMessage("An error just happened in me, blame the devs <:YoichiLol:701312070880329800>").queue()
             }
 
             override fun onResponse(call: Call, response: Response) {
+                if (response.code != 200) {
```

Comment:
What about the 429 error?

Reply:
I was going to add that but i don't know if its necessary or not

Reply:
Ill add it